ai/src/data_manager/time_series_manager.py

```diff
@@ -154,34 +117,15 @@ def get_interpretations_for_interval(self):
     def get_shape(self):
         return np.shape([0] * self.get_window_span() * len(self.fields))
 
-    @staticmethod
-    def flatten_and_normalize_window(current_window):
-        result_array = []
-        for col in current_window:
-            denominator = np.linalg.norm(current_window[col])
-            denominator = 1 if denominator == 0 else denominator
-
-            norm_rows = current_window[col] / denominator
-
-            for row in norm_rows:
-                result_array.append(row)
-        return np.nan_to_num(result_array)
-
-    def get_current_window(self):
-        with self.table_lock:
-            # This will get the nearest previous index that matches the timestamp,
-            # so we don't need to specify the timestamps exactly
-            start_index = self.massive_table_filled.index.get_loc(
-                self.current_time - self.param.interval_secs, "ffill"
-            )
-            end_index = self.massive_table_filled.index.get_loc(
-                self.current_time, "ffill"
-            )
-
-            return (
-                self.massive_table_filled.iloc[start_index:start_index + 1]
-                if self.get_window_span() == 1 else
-                self.massive_table_filled.iloc[start_index:end_index])
+    def get_current_window(self) -> pd.DataFrame:
+        # This will get the nearest previous index that matches the timestamp,
+        # so we don't need to specify the timestamps exactly
+        start_index = self.massive_table_filled.index.get_loc(self.current_time, "ffill")
+        end_index = self.massive_table_filled.index.get_loc(self.current_time + self.param.interval_secs, "ffill")
+        return (
+            self.massive_table_filled.iloc[start_index:start_index + 1]
+            if self.get_window_span() == 1 else
+            self.massive_table_filled.iloc[start_index:end_index])
```

Comment:
We need to keep the `with self.table_lock:` in this function until I have my PR that removes the lock safely. Otherwise it could conflict with `AddData`

Reply:
Ok, I reverted the function to use the lock (and added the same mechanism in the other manager)